mflowgen/tile_array/constraints/constraints.tcl

```diff
@@ -0,0 +1,75 @@
+#=========================================================================
+# Design Constraints File
+#=========================================================================
+
+# This constraint sets the target clock period for the chip in
+# nanoseconds. Note that the first parameter is the name of the clock
+# signal in your verlog design. If you called it something different than
+# clk you will need to change this. You should set this constraint
+# carefully. If the period is unrealistically small then the tools will
+# spend forever trying to meet timing and ultimately fail. If the period
+# is too large the tools will have no trouble but you will get a very
+# conservative implementation.
+
+set clock_net  clk
+set clock_name ideal_clock
+
+create_clock -name ${clock_name} \
+             -period ${dc_clock_period} \
+             [get_ports ${clock_net}]
+
+# This constraint sets the load capacitance in picofarads of the
+# output pins of your design.
+
+set_load -pin_load $ADK_TYPICAL_ON_CHIP_LOAD [all_outputs]
+
+# This constraint sets the input drive strength of the input pins of
+# your design. We specifiy a specific standard cell which models what
+# would be driving the inputs. This should usually be a small inverter
+# which is reasonable if another block of on-chip logic is driving
+# your inputs.
+
+set_driving_cell -no_design_rule \
+  -lib_cell $ADK_DRIVING_CELL [all_inputs]
+
+# set_input_delay constraints for input ports
+#
+# - make this non-zero to avoid hold buffers on input-registered designs
+
+set_input_delay -clock ${clock_name} [expr ${dc_clock_period}/2.0] [all_inputs]
+
+# set_output_delay constraints for output ports
+
+set_output_delay -clock ${clock_name} 0 [all_outputs]
+
+# Make all signals limit their fanout
+
+set_max_fanout 20 $dc_design_name
+
+# Make all signals meet good slew
+
+set_max_transition [expr 0.25*${dc_clock_period}] $dc_design_name
+
+#set_input_transition 1 [all_inputs]
+#set_max_transition 10 [all_outputs]
+
+# Ensure that no buffers get inserted on abutted nets
+# First get all nets connected to CGRA tiles
+set tile_nets [get_nets -of_objects [get_cells -filter {ref_lib_cell_name =~ *Tile*}]]
+# Now iterate over collection and check if net connects 
+# only 2 tiles. If it does, don't try to route it.
+foreach_in_collection net $tile_nets {
+  set connected_cells [get_cells -of_object $net]
+  set connected_tiles [filter_collection $connected_cells {ref_lib_cell_name =~ *Tile*}]
+  set num_connected_tiles [sizeof_collection $connected_tiles]
+  if {($num_connected_tiles > 1) && ([compare_collections $connected_cells $connected_tiles] == 0)} {
+    query_objects $net
+    set_dont_touch $net true
+  }
+}
+
+# This can catch nets connected to external ports of tile_array.
+# We must ensure that these nets can be touched
+set ext_port_nets [get_nets -of_objects [get_ports]]
+query_objects $ext_port_nets
+set_dont_touch $ext_port_nets false
```

Comment:
I'm guessing none of these would connect > 1 tiles though, right? Maybe just a precautionary thing...

Reply:
That was my initial thought, but it turns out they do. the clock, reset and config inputs branch out to all of the columns in the array. Without this, those nets were being marked as dont_touch.